### PR TITLE
New page: the `valueAsDate` property for `htmlinputelement`

### PR DESCRIPTION
**Pull request by estelle**
also fixes a typo for #35560
Part of https://github.com/mdn/content/issues/35544

### CODE REVIEW

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+<label for="date">Pick a date and time:</label>
+
+<input name="date" id="date" type="week" />
+
+<pre id="log"></pre>
+```
+
+#### JavaScript
+
+When no date or time is selected, the empty string resolves to `NaN`. Each time a selection is made, a {{domxref("HTMLElement/change_event", "change")}} event is fired, updating the `<pre>` content showing the {{DOMXref("HTMLInputElement.value")}} of the form control compared to that value as a number.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:41]:
```suggestion
When no date or time is selected, the empty input resolves to `null`. Each time a selection is made, a {{domxref("HTMLElement/change_event", "change")}} event is fired, updating the `<pre>` content showing the {{DOMXref("HTMLInputElement.value")}} of the form control compared to that value as a date.
```

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+browser-compat: api.HTMLInputElement.valueAsDate
+---
+
+{{ APIRef("HTML DOM") }}
+
+The **`valueAsDate`** property of the {{DOMxRef("HTMLInputElement")}} interface represents the current value of the {{htmlelement("input")}} element as a {{jsxref("Date")}}, or `null` if conversion is not possible.
+
+This property can also be set directly, for example to set a default date based on some condition.
+
+Throws an `InvalidStateError` {{domxref("DOMException")}} if the control isn't date- or time-based.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:15]: This is copied from the HTML spec but is IMO unclear.

```suggestion
When accessing this property on an input that isn't date- or time-based, returns `null`. When setting this property on such an input, throws an `InvalidStateError` {{domxref("DOMException")}}.
```

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+slug: Web/API/HTMLInputElement/valueAsDate
+page-type: web-api-instance-property
+browser-compat: api.HTMLInputElement.valueAsDate
+---
+
+{{ APIRef("HTML DOM") }}
+
+The **`valueAsDate`** property of the {{DOMxRef("HTMLInputElement")}} interface represents the current value of the {{htmlelement("input")}} element as a {{jsxref("Date")}}, or `null` if conversion is not possible.
+
+This property can also be set directly, for example to set a default date based on some condition.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:13]:
```suggestion
This property can also be set directly, for example to set a default date based on some condition. If the provided value is neither `null` nor a `Date` object, a {{jsxref("TypeError")}} is thrown. If the provided value is `null` or an [invalid date](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date), the input value is set to the empty string.
```

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+}
+```
+
+#### Results
+
+{{EmbedLiveSample("Retrieving a date value", "", 100)}}
+
+### Parsing a `valueAsDate` value
+
+This example demonstrates parsing the `valueAsDate` value of an `<input>` of type {{HTMLElement("input/date", "date")}} as a {{jsxref("Date")}} object.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:68]: You are not "parsing" anything, though; the `valueAsDate` is already the "parsed" value. You probably meant "displaying" or "formatting".

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+#log {
+  height: 20px;
+  padding: 0.5rem;
+  background-color: #ededed;
+}
+```
+
+#### Results
+
+{{EmbedLiveSample("Parsing a `valueAsDate` value", "", 100)}}
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:117]:
```suggestion
{{EmbedLiveSample("Parsing a valueAsDate value", "", 100)}}
```

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+}
+```
+
+#### Results
+
+{{EmbedLiveSample("Retrieving a date value", "", 100)}}
+
+### Parsing a `valueAsDate` value
+
+This example demonstrates parsing the `valueAsDate` value of an `<input>` of type {{HTMLElement("input/date", "date")}} as a {{jsxref("Date")}} object.
```
Comment (estelle) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:68]:
```suggestion
This example demonstrates applying {{jsxref("Date")}} methods directly to the `valueAsDate` property of the `HTMLInputElement` interface applied to an `<input>` of type {{HTMLElement("input/date", "date")}}.
```
wordy but clearer

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md (hunk 4 above)
Comment (estelle) [reply files/en-us/web/api/htmlinputelement/valueasdate/index.md]: changed to:

> This example demonstrates applying {{jsxref("Date")}} methods directly to the `valueAsDate` property of the `HTMLInputElement` interface applied to an `<input>` of type {{HTMLElement("input/date", "date")}}.

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+## Value
+
+A {{jsxref("Date")}} object or `null` if a conversion is impossible.
+
+## Examples
+
+### Retrieving a date value
+
+This example demonstrates that the `valueAsDate` property with an `<input>` of type {{HTMLElement("input/week", "week")}}.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:25]:
```suggestion
This example demonstrates accessing the `valueAsDate` property on an `<input>` of type {{HTMLElement("input/week", "week")}}.
```

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  padding: 0.5rem;
+  background-color: #ededed;
+}
+```
+
+#### Results
+
+{{EmbedLiveSample("Retrieving a date value", "", 100)}}
+
+### Parsing a `valueAsDate` value
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:66]: Change this title? Not sure in what way

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+<label for="date2">Pick a date:</label>
+
+<input name="date2" id="date2" type="date" />
+
+<pre id="log"></pre>
+```
+
+#### JavaScript
+
+When no date is selected, the empty string resolves to `null`. Each time a selection is made, a {{domxref("HTMLElement/change_event", "change")}} event is fired, we print the selected date using the
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/valueasdate/index.md:84]: Using the...?

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md (hunk 8 above)
Comment (estelle) [reply files/en-us/web/api/htmlinputelement/valueasdate/index.md]: Using Date methods

File: files/en-us/web/api/htmlinputelement/valueasdate/index.md (hunk 9 above)
Comment (estelle) [reply files/en-us/web/api/htmlinputelement/valueasdate/index.md]: > We then populate the log with the date selected, formatted using the `Date` object's {{jsxref("Date.prototype.toLocaleDateString()", "toLocaleDateString()")}} method.

**Review by Josh-Cena — APPROVED** [review]
LGTM, thanks for also fixing the typo

**Pull request merged**